addons/overheating/functions/fnc_canSwapBarrel.sqf

```diff
@@ -12,14 +12,15 @@
  */
 #include "script_component.hpp"
 params["_unit","_weapon"];
-private ["_canSwapBarrel","_weaponBarrelClass"];
+
+//Check if weapon can have its barrel swapped. If not exit out of function
+if( !GVAR(enabled) && (getNumber (configFile >> 'CfgWeapons' >> _weapon >> QGVAR(allowSwapBarrel))) != 1) exitWith{false};
+
 //Get the classname of the spare barrel for the weapon
-_weaponBarrelClass = getText (configFile >> 'CfgWeapons' >> _weapon >> 'ACE_Overheating_barrelClassname');
+private _weaponBarrelClass = getText (configFile >> 'CfgWeapons' >> _weapon >> QGVAR(barrelClassname));
 //If the weapon has no defined classname then use the ACE one
 if(_weaponBarrelClass == "") then {
-    _weaponBarrelClass = 'ACE_SpareBarrel';
+    _weaponBarrelClass = "ACE_SpareBarrel";
 };
-//Check if the player has the barrel and the weapon can have its barrel swapped
-_canSwapBarrel = GVAR(enabled) && {_weaponBarrelClass in magazines _unit} && {getNumber (configFile >> 'CfgWeapons' >> _weapon >> 'ACE_Overheating_allowSwapBarrel') == 1};
-
-_canSwapBarrel;
\ No newline at end of file
+//If the player has the spare barrel then it can be swapped
+if (_weaponBarrelClass in magazines _unit) exitWith{true};
```

Comment:
You need to return false otherwise.

```sqf
if (_weaponBarrelClass in magazines _unit) exitWith{true}; 
false;
```

or simply

```sqf
(_weaponBarrelClass in magazines _unit)
```